IMAGE_RENDERING.md

```diff
@@ -26,78 +26,28 @@ To tackle this **_we look to [WebGL](https://developer.mozilla.org/en-US/docs/We
+**[zarr](https://zarr.readthedocs.io/en/stable/)** essentially stores matrices of data in chunks using a variety of compression methods but very flexibly, in that the data can be of "any size", in "any shape". In other words, **zarr** stores n-dimensional chunks of the matrix in compressed [file blobs](https://zarr.readthedocs.io/en/stable/spec/v2.html), each flattened in `[row major order](https://en.wikipedia.org/wiki/Row-_and_column-major_order)` (this is basically how `[numpy](https://numpy.org/)` stores data interally in memory as well). Taking a step back to see why this is helpful, let us note that **WebGL** uses flat arrays and offset parameters to know how to scan the data so it can be passed down to the shaders - therefore **zarr**'s internal use of flattened data before it reaches the client results in very good performance. Thus, we store each image array of the pyramid in `tileSize x tileSize` passes over the data in a **zarr** file for efficient use. For example, if you have a `1028x1028x3` image with `tileSize` equal to 128, you could think of the data as being converted into a `10 x 10 x 128 x 128 x 3` matrix (i.e the top `128x128` entries of the original image becomes a block, then the next `128x128` entries and so forth), and then stored in `row-major order` to be served to the client. Thus you could request tile `A x B` and, in one request, get the `A x B`th chunk of size `128 x 128 x 3`, already flattened. Another advantage to this is that we can store each channel separately, thereby making our individual requests smaller and our application better an switching channels on and off- to render one channel, you only have to request one channel's worth of data, instead of requesting all channels needed in a "normal" image format. `TIFF` files are stored somewhat similarly but rather using the `tiff` file format so all the data lives in one file and then **geotiff.js** makes range requests to get a tile's worth of flattened image data. All this is to say, if you have a method that can store image data in a way that the client can efficiently request a tile's worth of data in one request, you can use that as the "backend" for this project (for example, if you had a `javascript` `hdf5` reader, you could probably use that!).
+
+Second, **_we make use of [HTTP2](https://en.wikipedia.org/wiki/HTTP/2)_** to speed up the requests. Even though the requests are not particularly large (normally far less than 2MB per tile with a `512x512` tile size, 16bits of data), there are many of them and in normal HTTP, they block one another after a certain point. **HTTP2** more or less circumvents this blocking problem, essentially reducing the time needed to make a request by the number of requests made. A possible extension of this is to use [gRPC](https://en.wikipedia.org/wiki/GRPC) (which uses **HTTP2** under the hood together with [Protobuf](https://en.wikipedia.org/wiki/Protocol_Buffers)) but for now, **HTTP2** alone works well enough.
```

Comment:
I think we should not that we make sure of HTTP/2 in our demos, but we should probably note that it is not a requirement! In fact, we haven't tested what storing an image on s3 is like (to my knowledge). We haven't mentioned that we are just using cloud object storage either. We should mention somewhere "backends" for viv, maybe in a table: Google Cloud storage (supports HTTP/2 out of the box), S3, your own server!

Reply:
also HTTP --> HTTP/1. I think the convention is the same for HTTP/2.

Reply:
Ok, I have tested s3 a bit a while back and it performs comparably to GCS, albeit differently.  The top picture is GCS and bottom is s3 for the final request to a zarr array.  I can post these and say something like "We think HTTP/2 in general gives better performance but we have not done extensive testing."

<img width="551" alt="Screen Shot 2020-04-11 at 2 27 00 PM" src="https://user-images.githubusercontent.com/43999641/79051864-89163680-7c00-11ea-8bee-b4657042e18d.png">
<img width="551" alt="Screen Shot 2020-04-11 at 2 26 52 PM" src="https://user-images.githubusercontent.com/43999641/79051866-89aecd00-7c00-11ea-9e53-22cdd65884e2.png">